### PR TITLE
Link to GitHub discussions

### LINKED ISSUE #2256: Enable GitHub Discussions for this Repository
GitHub made their Discussions open Beta and I strongly believe that this repository could benefit from having this.

Why?
People like me feel kinda uncomfortable to use issues for asking basic questions regarding the software/tool in general and often don't like to use sites/services like IRC or similar.
Having a discussion here would help us to more easily ask questions since it now would have a dedicated place for it and as an added bonus could be used as a nice collection of answers to questions someone might have.

### PR DESCRIPTION
**Pull request by Andre601**
Updates links from the groups to the new discussions section.

Closes #2256

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -343,7 +343,7 @@ you're done. For specific instructions on a number of common hosts, see the
 
 ## Getting help
 
-To get help with MkDocs, please use the [discussion group], [GitHub issues] or
+To get help with MkDocs, please use the [GitHub discussions], [GitHub issues] or
 the MkDocs IRC channel `#mkdocs` on freenode.
```
Comment (d0ugal) [inline docs/index.md:347]: I suspect we can drop the IRC channel too. I used to idle there and it was never that active. I doubt it is now.

Comment (Andre601) [reply docs/index.md]: That's something I let @waylan decide, but yeah. The two or three times I was there was it pretty dead.

Comment (waylan) [reply docs/index.md:347]: I never look at the IRC channel. I would suggest removing any mention of it.

Comment (Andre601) [reply docs/index.md]: Removed them.

**Comment by Andre601** [comment]
I took a bit of time and made a quick redesign of the readme, that I believe should be used here.
At the very least should the landscape links get removed as that service seems to be kind-of dead right now ([Here](https://landscape.io/the-current-state-of-landscape.html) is an article from the author announcing the death of the service).

Here's an image of the redesigned Readme.
image

**Comment by waylan** [comment]
@Andre601 I think your changes to the readme are a great start. Feel free to include them in a separate PR.

**Pull request merged**